### ISSUE TITLE
Creating reference set when working with dozens of features

### ISSUE THREAD

**Issue by psuanm5030** [aothman/hiscore#6]
Thanks for this package - great stuff (also thanks for the youtube video presentation you did).  

Your example gives a very clear & simple demonstration of the capabilities and was very helpful, however I am struggling conceptually with the reference set creation.  How can I build my reference set when I have, for example, 50-70 features.  I definitely plan to break it down into a tree like you show (as every 5-10 features deals with a unique sub-area that I want to score and combine to an overall score).  I imagine the combinations of these features being very large and unwieldy, since the example you gave has the two features relative to one scoring range, versus having each feature relative to its own scoring range.  So 2 features yields 9 potential scores (3^2).  I presume 10 features would yield ~59K potential scores (3^10).  Right?  

So in summary, how should i deal with reference sets that have many features to score?  Do you know of any good examples using HiScore?

Thanks for your time!

**Comment by aothman** [comment]
Thanks for the kind words!

You are very right that scoring in this way suffers from the [Curse of Dimensionality](https://en.wikipedia.org/wiki/Curse_of_dimensionality). I had a professor in graduate school who claimed that he had a professor in graduate school who could "see" up to ten dimensions. But for us mortals, we generally lose touch of non-linear relationships after about three dimensions.

I would suggest splitting each score all the way down until you have subscores that contain no more than five (and ideally, no more than three) numerical input dimensions. These subscores should then return 0-100 values. Then, combine the subscores up a tree as in the README, two or more at a time, and have them also output 0 to 100 values. Since you're normalizing the output space, it is generally easiest to just combine them through a simple weighted average combination. But, you _could_ combine these subscores using `hiscore` if you want a non-linear response.
